spring-cloud-dataflow-audit/src/main/java/org/springframework/cloud/dataflow/audit/service/DefaultAuditRecordService.java

```diff
@@ -38,74 +41,101 @@
-
-	private final AuditRecordRepository auditRecordRepository;
-
-	private final ObjectMapper objectMapper;
-
-	public DefaultAuditRecordService(AuditRecordRepository auditRecordRepository) {
-		Assert.notNull(auditRecordRepository, "auditRecordRepository must not be null.");
-		this.auditRecordRepository = auditRecordRepository;
-		this.objectMapper = new ObjectMapper();
-		this.objectMapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
-	}
-
-	public DefaultAuditRecordService(AuditRecordRepository auditRecordRepository, ObjectMapper objectMapper) {
-		Assert.notNull(auditRecordRepository, "auditRecordRepository must not be null.");
-		Assert.notNull(objectMapper, "objectMapper must not be null.");
-		this.auditRecordRepository = auditRecordRepository;
-		this.objectMapper = objectMapper;
-	}
-
-	@Override
-	public AuditRecord populateAndSaveAuditRecord(AuditOperationType auditOperationType,
-			AuditActionType auditActionType,
-			String correlationId, String data, String platformName) {
-		Assert.notNull(auditActionType, "auditActionType must not be null.");
-		Assert.notNull(auditOperationType, "auditOperationType must not be null.");
-
-		final AuditRecord auditRecord = new AuditRecord();
-		auditRecord.setAuditAction(auditActionType);
-		auditRecord.setAuditOperation(auditOperationType);
-		auditRecord.setCorrelationId(correlationId);
-		auditRecord.setAuditData(data);
-		auditRecord.setPlatformName(platformName);
-		return this.auditRecordRepository.save(auditRecord);
-	}
-
-	@Override
-	public AuditRecord populateAndSaveAuditRecordUsingMapData(AuditOperationType auditOperationType,
-			AuditActionType auditActionType,
-			String correlationId, Map<String, Object> data, String platformName) {
-		String dataAsString;
-		try {
-			dataAsString = objectMapper.writeValueAsString(data);
-		}
-		catch (JsonProcessingException e) {
-			logger.error("Error serializing audit record data.  Data = " + data);
-			dataAsString = "Error serializing audit record data.  Data = " + data;
-		}
-		return this.populateAndSaveAuditRecord(auditOperationType, auditActionType, correlationId, dataAsString, platformName);
-	}
-
-	@Override
-	public Page<AuditRecord> findAuditRecordByAuditOperationTypeAndAuditActionTypeAndDate(
-			Pageable pageable,
-			AuditActionType[] actions,
-			AuditOperationType[] operations,
-			Instant fromDate,
-			Instant toDate) {
-		return this.auditRecordRepository.findByActionTypeAndOperationTypeAndDate(operations, actions, fromDate, toDate,
-				pageable);
-	}
-
-	@Override
-	public Optional<AuditRecord> findById(Long id) {
-		return this.auditRecordRepository.findById(id);
-	}
+    private static final Logger logger = LoggerFactory.getLogger(DefaultAuditRecordService.class);
+
+    private final AuditRecordRepository auditRecordRepository;
+
+    private final ObjectMapper objectMapper;
+    private final ArgumentSanitizer sanitizer;
```

Comment:
**[NIT]** spacing inconsistent w/ other member vars. I am ok either way w/ or w/o the space in between the vars as long as they are consistent.

Reply:
Done